Suggest-guide-overhaul (#1)
Biskit1943/Encrypted-LVM-Arch: README.md

[Download](https://www.archlinux.org/download/ "Arch download")

> latest Iso

Reviewer: ```suggestion
the latest Iso
```

## TL;DR
Before you tackle this ~1h guide it should be mentioned that you can install

> arch in a guided manner with the

Reviewer: ```suggestion
Arch in a guided manner with the
```

[archinstall](https://wiki.archlinux.org/title/archinstall) helper library. To

> get an idea of this 'easy arch install' visit their

Reviewer: ```suggestion
get an idea of this 'easy Arch install' visit their
```

> [docs](https://python-archinstall.readthedocs.io/en/latest/installing/guided.html)

Reviewer: ```suggestion
[docs](https://python-archinstall.readthedocs.io/en/latest/installing/guided.html).
```

> Lets further add that you could with the same libary install arch directly from

Reviewer: ```suggestion
Lets further add that you could with the same libary install Arch directly from
```

a configuration file. If you decide to use the guided way you can start the

> guide at section `After first reboot`

Reviewer: You can actually link to the anchor in this markdown by using:
```suggestion
guide at section [After first reboot](#After-first-reboot)
```

> While automated tools like archinstall streamline the installation process,

Reviewer: Love the block!

going through this manual guide offers you the advantage of a deeper
understanding of your system's architecture. This guide equips you with valuable
insights into component configuration, allowing for a level of control and
customization that's hard to achieve otherwise. Additionally, should your system
become 'bricked' or unresponsive, this knowledge could be invaluable for
troubleshooting and recovery. Consider this manual process an investment in your
own skills and system mastery.

## Partitioning the Harddrive

> This guide covers both EFI and Legacy boot modes.

Reviewer: Add newline after header

> # Pre-requisites

Reviewer: ```suggestion
### Pre-requisites
```

> 1. For an EFI system, ensure that Secure Boot is disabled in your system BIOS.

Reviewer: This should be true no matter what type of boot you use.
Maybe put `Secure Boot` in ticks like here?

2. If you are using VirtualBox, it's recommended to avoid EFI mode as it can
   introduce complications. For those who still wish to proceed with EFI on
   VirtualBox, consult this [workaround](https://askubuntu.com/a/573672).

> Determine if your system has booted in EFI or Legacy mode. Run the following

Reviewer: ```suggestion
To determine if your system has booted in EFI or Legacy mode, run the following
```

command:
```bash
$ ls /sys/firmware/efi/efivars
```
If the directory does not exist, you have booted into Legacy mode.

> # Identifying the Target Drive

Reviewer: ```suggestion
### Identifying the target drive
```

sda      8:0     0     10G   0  disk
sr0     11:0     1    523M   0   rom  /run/archiso/bootmnt

> ```

Reviewer: Add a newline after a codeblock

> Here, `sda` is the target drive where we intend to install Arch Linux. Next, we'll

Reviewer: Since you were confused by this, maybe add additional example for nvmes?

create two partitions: one for the bootloader and another for LVM.

> # Creating Partitions

Reviewer: ```suggestion
### Creating partitions
```

> Run gdisk to interactively partition your target drive:

Reviewer: Keywords like executables should be put in ticks
```suggestion
Run `gdisk` to interactively partition your target drive:
```

[A sidenode from the arch wiki:](https://wiki.archlinux.org/title/partitioning)

> "A suggested size for /boot is 200 MiB unless you are using EFI system partition

Reviewer: Citing can be archived by using `>`:
```suggestion
> A suggested size for /boot is 200 MiB unless you are using EFI system partition
> as /boot, in which case at least 300 MiB is recommended. If you want to install
> multiple kernels or want to be future proof, you can use 1 GiB to be on the safe
> side. 
```

as /boot, in which case at least 300 MiB is recommended. If you want to install
multiple kernels or want to be future proof, you can use 1 GiB to be on the safe
side. "

```bash
$ gdisk /dev/sda

> ```

Reviewer: Add a newline after a codeblock

If neccesary delete old partitions first.
Then start with the boot partition.
```
Command (? for help): n
Partition number (1-128, default 1): <enter>
First sector (34-20971486, default = 2048) or {+-}size{KMGTP}: <enter>

> Last sector (2048-20971486, default = 20971486) or {+-}size{KMGTP}: +M # Don't hesitate to pick 2GB here.

Reviewer: ```suggestion
Last sector (2048-20971486, default = 20971486) or {+-}size{KMGTP}: +1024M # Don't hesitate to pick 2048M (2GB) here.
```

Hex code or GUID (L to show codes, Enter = 8300): ef00  # for EFI, ef02 for LEGACY
```

> Continue with the LVM partition

Reviewer: Add a newline below the codeblock

```
Command (? for help): n
Partition number (2-128, default 2): <enter>
First sector (34-20971486, default = 20973568) or {+-}size{KMGTP}: <enter>
Last sector (20973568-20971486, default = 20971486) or {+-}size{KMGTP}: <enter>
Hex code or GUID (L to show codes, Enter = 8300): 8e00
```

> # (Optional) Securely Wiping the Drive 

Reviewer: ```suggestion
### (Optional) Securely wiping the drive 
```

If you are using a new disk, this step is not mandatory but can be performed for added security.
```bash
$ shred -v -n 1 /dev/sda  # use -n 1 for a single pass instead of the default 3 passes
```

> # Setting Up Encrypted LVM

Reviewer: ```suggestion
## Setting up the encrypted LVM
```

Before configuring the encrypted LVM (Logical Volume Manager), it's essential to

> load the required kernel module, dm-crypt. This module enables disk encryption

Reviewer: ```suggestion
load the required kernel module, `dm-crypt`. This module enables disk encryption
```

on Linux.

> Load the dm-crypt Kernel Module 

Reviewer: ```suggestion
Load the `dm-crypt` Kernel Module 
```

```bash
$ modprobe dm-crypt
```

> the next step is to encrypt the LVM partition. In this guide example, the LVM

Reviewer: ```suggestion
The next step is to encrypt the LVM partition. In this guides example, the LVM
```

partition is /dev/sda2.
```
$ cryptsetup -c aes-xts-plain64 -y -s 512 luksFormat /dev/sda2
  >> YES  # Confirm that all Data will be overriden
  Enter passphrase: >> *Secred Password*
  Verify passphrase: >> *Secred Password*

>  ```

Reviewer: Add a newline after a codeblock

After successfully encrypting your LVM partition, you'll need to unlock it to
proceed with the installation. To unlock your encrypted LVM partition, use the
following command:

```bash
$ cryptsetup luksOpen /dev/sda2 lvm  # this will mount the device under /dev/mapper/lvm

> ```

Reviewer: Add a newline after a codeblock

Note: If you ever face an issue that prevents you from accessing your hard
drive, you can use this command from a recovery environment to unlock the
partition.

After unlocking the encrypted partition, it's essential to set up Logical Volume
Management (LVM) and initialize the file systems for the installation.

Initialize `/dev/mapper/lvm` as a physical volume:

> 

Reviewer: To be consistent throughout the guide you should either always add a newline *before* a code block or never.
Maybe consider to use something like prettier to format the file?

```bash
$ pvcreate /dev/mapper/lvm

> ```

Reviewer: Add a newline after a codeblock

> Create a volume group named `main`: all the subsequent logical volumes (root,

Reviewer: ```suggestion
Create a Volume Group (VG) named `main`: all the subsequent logical volumes (root,
```

swap, home) will be part of this volume group.

```bash
$ vgcreate main /dev/mapper/lvm # "main" is the name of your volume group.

> ```

Reviewer: Add a newline after a codeblock

> Create root and swap Logical Volumes: These logical volumes will reside inside

Reviewer: ```suggestion
Create `root` and `swap` Logical Volumes: These logical volumes will reside inside
```

> the "main" volume group.

Reviewer: ```suggestion
the `main` volume group.
```

> Note: Consider using a swap file 

Reviewer: ```suggestion
Note: Consider using a swap file.
```

If you want the ability to hibernate or open files that are
larger then your RAM, then your swap space should be at least as large as your

> existing RAM. This is because during hibernation, the contents of RAM are

Reviewer: I would no longer recommend sizes larger than 16GB. More is not necessary, imho. Maybe add a hint, that up until 16GB RAM you use the same and if you have more RAM then use whatever you like.

written to the swap space.

```bash

> $ lvcreate -L XGB -n swap main 

Reviewer: You previously added a comment to hint, that `main` is the name of the lvm, maybe also add it here :)

$ ping -c3 www.archlinux.org
```

If you are not connected to the Internet and wish to use Wi-Fi, you'll need to
identify your wireless network interface and connect using wpa_supplicant. Use
the following command to list network interfaces and find your wireless network
device:
```bash
$ ip a
```

Then connect through wpa_supplicant.

> Figure out you <wifi-name> with `ip a`.

Reviewer: ```suggestion
Figure out your <wifi-name> with `ip a`.
```

```bash
$ wpa_supplicant -B -i <wifi-name> -c <(wpa_passphrase "Your_SSID" Your_passphrase) && dhclient <wifi-name>
```

> For more details, consult the [official documentation](https://wiki.archlinux.org/title/Wpa_supplicant) 

Reviewer: ```suggestion
For more details, consult the [official documentation](https://wiki.archlinux.org/title/Wpa_supplicant).
```

If you encounter issues or prefer a different method for connecting to the Internet, consult the

> [Arch Linux Installation Guide.](https://wiki.archlinux.org/index.php/Installation_guide#Connect_to_the_Internet)

Reviewer: ```suggestion
[Arch Linux Installation Guide.](https://wiki.archlinux.org/index.php/Installation_guide#Connect_to_the_Internet).
```

If you're facing issues with obtaining an IP address, you can manually trigger
the DHCP client as follows:

```bash
$ dhcpcd
```

> ## Configure Mirror List (Optional)

Reviewer: ```suggestion
## Configure the mirror list (Optional)
```

Author: did put `Configure the Mirror List` and streamlined all the headings to be uppercase except words like `the`.
In other words I go with "Title Case" not "Sentence case" in all headings

Note: The need to adjust the mirror list can depend on various factors, such as
your geographical location, network speed requirements, and specific use-cases.
For most users, the default mirror list should suffice, especially given that
Arch Linux in 2023 utilizes a ranked system for mirrors. This makes manual
intervention less necessary. However, if you experience slower-than-expected
download speeds and are not ( a digital nomad ) frequently changing your
geographical location, you may want to optimize the mirror list.

Before making any changes, backup your current mirror list:
```bash
$ cp /etc/pacman.d/mirrorlist /etc/pacman.d/mirrorlist.bak
```

To use mirrors from a specific country (replace XX with your country's domain
code, e.g., de for Germany):

```bash
$ grep -E  ".*\.XX.*$" /etc/pacman.d/mirrorlist.bak > /etc/pacman.d/mirrorlist
```
Check to make sure your mirror list now only contains entries for your chosen country:

```bash
$ cat /etc/pacman.d/mirrorlist
```

> ## Installing the Base System

Reviewer: ```suggestion
## Installing the base system
```

Author: I go with "Title Case" not "Sentence case" in all headings

Optional packages are:
1. [intel-ucode](https://wiki.archlinux.org/index.php/Microcode),
  for Intel processors.
2. [amd-ucode](https://wiki.archlinux.org/index.php/Microcode),
  for AMD processors.
3. [wpa_supplicant](https://wiki.archlinux.org/index.php/WPA_supplicant),

>   if you're using WLAN

Reviewer: ```suggestion
  if you're using WIFI
```

4. [linux-lts](https://archlinux.org/packages/core/x86_64/linux-lts/),

>   if you want a secondary stable LTS linux kernel for backup usecases

Reviewer: ```suggestion
  if you want a secondary stable LTS linux kernel for backup usecases or as an alternative to the rolling release
```

5. Other packages may be essential based on your specific needs.

Install everything you need for your System with (`[]` are optional dependencies):

```bash
$ pacstrap /mnt base linux linux-firmware grub efibootmgr base-devel vim dhcpcd \
cryptsetup lvm2 which inetutils man-db man-pages sudo curl gzip unzip \
[intel-ucode] [linux-lts] [wpa_supplicant] [usbutils] [git] [diffutils] \
[jfsutils] [less] [logrotate] [wget] [neovim]
```

After the installation finished we generate the Filesystem Table (fstab) to
ensure that all filesystems are correctly mounted during boot

```bash
$ genfstab -U -p /mnt >> /mnt/etc/fstab
```

This will generate the fstab file using UUIDs for identifying partitions and

> append the generated entries into /mnt/etc/fstab. 

Reviewer: ```suggestion
append the generated entries into `/mnt/etc/fstab`. 
```

> Verifying the fstab file

Reviewer: ```suggestion
Verifying the fstab file:
```

$ echo YourHostName >> /etc/hostname
```

> Set your system's time zone by creating a symbolic link from the appropriate

Reviewer: ```suggestion
Set your systems time zone by creating a symbolic link from the appropriate
```

> zoneinfo file to /etc/localtime. Replace Region and City with your specific time

Reviewer: ```suggestion
zoneinfo file to /etc/localtime. Replace `Region` and `City` with your specific time
```

zone details. E.g. Region (Europe) and City (Berlin) for Germany.

```bash
$ ln -sf /usr/share/zoneinfo/Region/City /etc/localtime
```

> Run the hwclock command to generate /etc/adjtime and synchronize the hardware

Reviewer: ```suggestion
Run the `hwclock` command to generate `/etc/adjtime` and synchronize the hardware
```

clock to the system clock:

```bash
$ hwclock --systohc
```

> Edit the /etc/locale.gen file to uncomment the locales you want to generate.

Reviewer: ```suggestion
Edit the `/etc/locale.gen` file to uncomment the locales you want to generate.
```

> Then generate it.

Reviewer: ```suggestion
Then generate it:
```

$ locale
```

> The mkinitcpio.conf file is used to configure the initial RAM filesystem

Reviewer: ```suggestion
The `mkinitcpio.conf` file is used to configure the initial RAM filesystem
```

> (initramfs) in Arch Linux. The MODULES and HOOKS lines in this file specify

Reviewer: ```suggestion
(initramfs) in Arch Linux. The `MODULES` and `HOOKS` lines in this file specify
```

  *Secred Password*
```

Enable `dhcpcd` so you have wired internet after reboot.

```bash
$ systemctl enable dhcpcd.service
```

> If you need wifi internet after reboot you could revisit the wpa_supplicant

Reviewer: ```suggestion
If you need WIFI internet after reboot you could revisit the wpa_supplicant
```
Either use `wifi` all the time or `WIFI` but be consistent

> command above to enable that manually once rebooted. Later in this guide nmcli

Reviewer: ```suggestion
command above to enable that manually once rebooted. Later in this guide `nmcli`
```

> is itroduce as the 'automatic way of wi-fi connecting'.

Reviewer: ```suggestion
is itroduce as the 'automatic way of WIFI connecting'.
```

> Configure your Grub bootloader

Reviewer: ```suggestion
Configure your Grub bootloader:
```

Ignore them this ist because you are in a chroot env.

> Now exit and reboot into your new system

Reviewer: ```suggestion
Now exit and reboot into your new system:
```

```bash
$ exit
$ umount /mnt/{boot,home,}
$ reboot
```
## After first reboot

> Login and create a non-root user

Reviewer: ```suggestion
Login and create a non-root user:
```

```bash
$ useradd -m -g users -s /bin/bash your-username
$ passwd your-username
  *Secred Password*
  *Secred Password*
$ EDITOR=vim visudo
  >> Uncomment %wheel ALL=(ALL) ALL
  >> save and exit (:wq)
$ gpasswd -a your-username wheel
```

> # (Optional) connect to Wifi via NetworkManager

Reviewer: ```suggestion
## (Optional) connect to the WIFI via NetworkManager
```

If you dont have Internet you could plug in your wired connection or connect
once again manually via wpa_supplicant.

> First figure out you <wifi-name> with `ip a` then use the following command:

Reviewer: ```suggestion
First figure out you `<wifi-name>` with `ip a` then use the following command:
```

```bash
$ wpa_supplicant -B -i <wifi-name> -c <(wpa_passphrase "Your_SSID" Your_passphrase) && dhcpcd <wifi-name>

> ```

Reviewer: Newline

> If you have not already you can install, enable and start NetworkManager. Note

Reviewer: ```suggestion
If you have not already, you can install, enable and start NetworkManager (Note
```

> that the package name is plain lowercase and the service name is PascalCase

Reviewer: ```suggestion
that the package name is plain lowercase and the service name is PascalCase):
```

```bash
$ pacman -s networkmanager
$ systemctl enable NetworkManager
$ systemctl start NetworkManager
```

> Connect to your Wifi

Reviewer: ```suggestion
Connect to your WIFI
```

```bash
$ nmcli device wifi connect 'Your_SSID' password 'Your_Password'
```

> To make sure your system automatically reconnects to this Wi-Fi network whenever

Reviewer: ```suggestion
To make sure your system automatically reconnects to this WIFI network whenever
```

it's available, you can modify the connection settings to enable auto-connect.

```bash
$ nmcli connection show
```

Then if desired, set the connection to auto-connect and/or disable powersave to
bolster a stable wifi connection and restart:

```bash
$ nmcli connection modify 'Your_Connection_Name' connection.autoconnect yes
$ nmcli connection modify 'Your_Connection_Name' 802-11-wireless.powersave 2
$ systemctl restart NetworkManager
```

> # Install useful Services

Reviewer: ```suggestion
## Install useful services
```

> 1. acpid, a modue that handles ACPI events (Such as closing the lid of

Reviewer: ```suggestion
1. `acpid,` a modue that handles ACPI events (Such as closing the lid of
```

a notebook)

> 2. dbus, message System for inter-process communication

Reviewer: ```suggestion
2. `dbus,` message System for inter-process communication
```

$ pacman -S acpid dbus avahi cups cronie
$ systemctl enable acpid
$ systemctl enable avahi-daemon
$ systemctl enable cups.service
$ systemctl enable cronie
```

Now your base system is ready, you can now install your preferred
Desktop or use arch from the command Line

> # (Optional) Install AUR package manager.

Reviewer: ```suggestion
## (Optional) Install an AUR package manager.
```

> There are various package managers e.g. yay but here we install RUA a build tool

Reviewer: ```suggestion
There are various package managers e.g. `yay` but here we install `rua` as a build tool
```

> for ArchLinux, AUR.

Reviewer: ```suggestion
for the AUR.
```
The `A` stands for Arch already
